### PR TITLE
Python collections wrapper and passage retrieval setup

### PR DESCRIPTION
**Pull request by emmileaf**
Keeping an open PR with initial setup for iterating over collections with Pyjnius and segmenting documents, as well as discussion for adapting this to generally address [#602](https://github.com/castorini/anserini/issues/602).

### CODE REVIEW

**Comment by emmileaf** [comment]
Currently the setup uses the java document generator classes for collection-specific content parsing logic. This is more interlaced with other parts of the code such as IndexCollection:


https://github.com/castorini/anserini/blob/32e01911f34766b1d87bbf3a3c5efc31ecb4f206/src/main/java/io/anserini/index/generator/LuceneDocumentGenerator.java#L75-L79

An alternative to this approach could be providing a Python interface for only the collections classes, and skip using generator classes to convert `SourceDocument` into `LuceneDocument`? This would move the content parsing and HTML cleaning logic over to be addressed or re-written in Python instead.

File: src/main/python/segment_collection/utils.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import jnius_config
```
Comment (lintool) [inline src/main/python/segment_collection/utils.py]: And you're planning on moving common code to something like `src/main/python/io/anserini/collection/`? We probably want to mirror the java package structure?

File: src/main/python/segment_collection/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+### Segmenting collection
```
Comment (lintool) [inline src/main/python/segment_collection/README.md]: Let's call this `passage_retrieval`?
So move everything to `src/main/python/passage_retrieval/`?

**Comment by emmileaf** [comment]
PR changes:

- moved to  [`src/main/python/passage_retrieval`](https://github.com/emmileaf/anserini/tree/python-collection/src/main/python/passage_retrieval)
- used Python enums for specifying collection and generator classes
- factored out a common collections wrapper into [`passage_retrieval/collection`](https://github.com/emmileaf/anserini/tree/python-collection/src/main/python/passage_retrieval/collection) for now, which can support usage similar to:

```
collection = Collection(collection_class, input_path)

for (i, fs) in enumerate(collection.segments):
    for (i, doc) in enumerate(fs):
         foo(doc)
```

[`passage_retrieval/collection`](https://github.com/emmileaf/anserini/tree/python-collection/src/main/python/passage_retrieval/collection) is what I can move into `src/main/python/io/anserini/collection/` for later…  In terms of mirroring the java package structure, are we expecting to have other subdirectories fall into `src/main/python/io/anserini/` in the future? @lintool

File: src/main/python/passage_retrieval/collection_iterator.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    logger.info("# Final Counter Values");
+    logger.info("indexable:     {:12d}".format(collection.counters.indexable.value))
+    logger.info("unindexable: {:12d}".format(collection.counters.unindexable.value))
+    logger.info("skipped:     {:12d}".format(collection.counters.skipped.value))
+    logger.info("errors:      {:12d}".format(collection.counters.errors.value))
+    
+    logger.info("Total duration: %s", str(datetime.timedelta(seconds=elapsed)))
+    
+
+#IterCollection('C:/cygwin64/home/Emily/usra/collection/disk45',
```
Comment (lintool) [inline src/main/python/passage_retrieval/collection_iterator.py]: probably want to remove this?

**Comment by lintool** [comment]
> passage_retrieval/collection is what I can move into src/main/python/io/anserini/collection/ for later… In terms of mirroring the java package structure, are we expecting to have other subdirectories fall into src/main/python/io/anserini/ in the future? @lintool

Yes, please move over? I anticipate that we'll gradually mirror and "export" Java classes as Python gets more use. For example, we probably want to expose the topic readers in Python also... I can imagine doing learning-to-rank in Python also...

Let's do the `src/main/python/io/anserini/collection/` move, and we should merge this...

**Pull request merged**